src/loc/VoidLocReplaceExistingButton.tsx

```diff
@@ -8,60 +8,68 @@ import { useCommonContext } from "../common/CommonContext";
-                setReplacerLocIdError("Cannot be replaced by a VOID LOC");
-            } else if(replacerLoc.locType !== locData?.locType) {
-                setReplacerLocIdError(`Cannot be replaced by a LOC of type: ${replacerLoc.locType}`);
-            } else {
-                setReplacerLocIdError(undefined);
-                setCall(() => (callback: CallCallback) => mutateLocState(async current => {
-                    if(signer && (current instanceof OpenLoc || current instanceof ClosedLoc || current instanceof ClosedCollectionLoc)) {
-                        return current.legalOfficer.voidLoc({
-                            reason,
-                            replacer: locId,
-                            signer,
-                            callback,
-                        });
+                if (replacerLocId === locData?.id.toDecimalString()) {
+                    setReplacerLocIdError("Cannot be replaced by itself");
+                } else {
+                    const replacerLoc = await api!.queries.getLegalOfficerCase(locId);
+                    if (!replacerLoc) {
+                        setReplacerLocIdError("LOC not found on chain");
+                    } else if (replacerLoc.voidInfo !== undefined) {
+                        setReplacerLocIdError("Cannot be replaced by a VOID LOC");
+                    } else if (replacerLoc.locType !== locData?.locType) {
+                        setReplacerLocIdError(`Cannot be replaced by a LOC of type: ${ replacerLoc.locType }`);
                     } else {
-                        return current;
+                        setReplacerLocIdError(undefined);
+                        await mutateLocState(async current => {
+                            if (signer && (current instanceof OpenLoc || current instanceof ClosedLoc || current instanceof ClosedCollectionLoc)) {
+                                return current.legalOfficer.voidLoc({
+                                    reason,
+                                    replacer: locId,
+                                    signer,
+                                    callback,
+                                });
+                            } else {
+                                return current;
+                            }
+                        });
                     }
-                }));
+                }
             }
         }
-        }
     }, [ replacerLocId, api, reason, locData, mutateLocState, signer ]);
 
     const clearAndClose = useCallback(() => {
+        clearSubmissionState();
         setReason("");
         setReplacerLocId("");
         setReplacerLocIdError(undefined);
         setVisible(false);
-    }, [ setReason, setReplacerLocId, setReplacerLocIdError, setVisible ]);
+    }, [ clearSubmissionState, setReason, setReplacerLocId, setReplacerLocIdError, setVisible ]);
+
+    useEffect(() => {
+        if (extrinsicSubmissionState.isSuccessful()) {
+            clearAndClose();
```

Comment:
Same remark as for void (move to callback etc.).

Reply:
Done as suggested.